api/hasura/actions/_handlers/createCircle.ts

```diff
@@ -66,15 +67,24 @@ async function handler(req: VercelRequest, res: VercelResponse) {
 function createCircle(
   circleInput: z.infer<typeof createCircleSchemaInput>,
   userAddress: string,
+  userProfileId: number,
   coordinapeAddress: string
 ) {
   return async (fileName: string) => {
-    return await mutations.insertCircleWithAdmin(
+    const circle = await mutations.insertCircleWithAdmin(
       circleInput,
       userAddress,
       coordinapeAddress,
       fileName
     );
+
+    await mutations.insertInteractionEvents({
```

Comment:
@crabsinger i just noticed this while working on some related code... for performance, i think it would be ideal to bundle this with the request above rather than create another request-response loop. no need to fix this now, just flagging the general pattern.

Reply:
@levity yes it would be more performant... but not sure how to batch these since we still want the sql logic separated in different mutations? ... 

do you see a clean pattern that would support this?

Idon't know if there is a Hasura "batch" syntax or something...

Reply:
> we still want the sql logic separated in different mutations?

not sure what you mean, but maybe we don't? there's a standard GraphQL syntax for doing multiple queries or mutations in a single request; you can see an example [here](https://github.com/coordinape/coordinape/blob/main/api/hasura/cron/recoverTransactions.ts#L150)